### PR TITLE
Clean up actions

### PR DESCRIPTION
**Pull request by btkostner**
- Moves all of the "CI" stuff to a single "CI" workflow
- Only builds the project once for "CI" (speeeeeeed)
- Cleans up names of all the jobs and steps

### CODE REVIEW

**Comment by btkostner** [comment]
This will need to fix the "master" branch CI protection. Just let me know once approved.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+      - name: Upload
+        uses: actions/upload-artifact@v1
+        with:
+          name: vala-lint
+          path: ./
+
+  Test:
+    name: Test Build
+    runs-on: ubuntu-latest
```
Comment (benwaffle) [inline .github/workflows/ci.yml]: I think `runs-on` is not needed if you specify `container`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    needs: Build
+
+    container:
+      image: elementary/docker:unstable
```
Comment (benwaffle) [inline .github/workflows/ci.yml]: can just do `container: elementary/docker:unstable`

File: .github/workflows/ci.yml (hunk 1 above)
Comment (btkostner) [reply .github/workflows/ci.yml]: Nope. Still need `runs-on`.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (btkostner) [reply .github/workflows/ci.yml]: `container` needs to be an object, not a string. You get both of these errors in the workflow visual editor on GitHub.

Comment (benwaffle) [reply .github/workflows/ci.yml]: I was just going off of the docs https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idcontainer

Comment (btkostner) [reply .github/workflows/ci.yml]: Something is wrong then...

Screenshot from 2019-10-11 11 09 19

**Pull request merged**